src/api/tigl.cpp

```diff
@@ -2095,31 +2100,18 @@ TIGL_COMMON_EXPORT TiglReturnCode tiglWingSegmentPointGetComponentSegmentEtaXsi(
     
     try {
         tigl::CCPACSConfigurationManager& manager = tigl::CCPACSConfigurationManager::GetInstance();
-        tigl::CCPACSConfiguration& config = manager.GetConfiguration(cpacsHandle);
+        tigl::CCPACSConfiguration& config         = manager.GetConfiguration(cpacsHandle);
+        const auto& uidMgr                        = config.GetUIDManager();
 
-        // search for component segment
-        int nwings = config.GetWingCount();
-        for (int iwing = 1; iwing <= nwings; ++iwing) {
-            tigl::CCPACSWing& wing = config.GetWing(iwing);
-            try {
-                tigl::CCPACSWingComponentSegment & compSeg = (tigl::CCPACSWingComponentSegment &) wing.GetComponentSegment(componentSegmentUID);
-                compSeg.GetEtaXsiFromSegmentEtaXsi(segmentUID, segmentEta, segmentXsi, *eta, *xsi);
-                return TIGL_SUCCESS;
-            }
-            catch (const tigl::CTiglError& err){
-                if (err.getCode() == TIGL_UID_ERROR) {
-                    continue;
-                }
-                else {
-                    throw;
-                }
-            }
-            
+
+        if (!uidMgr.IsUIDRegistered(componentSegmentUID) || !uidMgr.IsType<tigl::CCPACSWingComponentSegment>(componentSegmentUID)) {
```

Comment:
Just a comment - not related to your changes: The current API of the uidManager makes coding pretty verbose. Maybe we should change the interface to the following:

```cpp
template <typename T>
T* UidManager::ResolveObject<T>(std::string uid)
```
If UID id not available or the type does not match, it will return a nullptr instead of throwing an exception.